sbt-metals/src/main/scala/scala/meta/sbt/MetalsPlugin.scala

```diff
@@ -0,0 +1,189 @@
+      configs.flatMap { config =>
+        inConfig(config)(
+          Seq(
+            metalsBuildInfo := Metals.metalsBuildInfoTask.value,
+            metalsWriteBuildInfo := Metals.metalsWriteBuildInfoTask.value
+          )
+        )
+      } ++ Seq(
+        // without config scope it will aggregate over all project dependencies
+        // and their configurations
+        metalsWriteBuildInfo := Def.taskDyn {
+          val depsAndConfigs = ScopeFilter(
+            inDependencies(ThisProject),
+            inConfigurations(configs: _*)
+          )
+          metalsWriteBuildInfo.all(depsAndConfigs)
+        }.value
+      )
+    }
+  }
+  import autoImport._
+
+  override def projectSettings =
+    metalsSettings(Compile, Test)
+
+  override def globalSettings = Seq(
+    commands ++= Seq(
+      Metals.semanticdbEnable,
+      Metals.metalsSetup
+    ),
+    // without project scope it will aggregate over all projects
+    metalsWriteBuildInfo := {
+      metalsWriteBuildInfo.all(ScopeFilter(inAnyProject)).value
+      streams.value.log.info("🤘 Metals is ready, time to rock!")
+    }
+  )
+}
+
+object Metals {
+
+  def metalsBuildInfoTask: Def.Initialize[Task[Map[String, String]]] =
+    Def.task {
+      Map(
+        "sources" -> sources.value.distinct.mkString(File.pathSeparator),
+        "unmanagedSourceDirectories" ->
+          unmanagedSourceDirectories.value.distinct
+            .mkString(File.pathSeparator),
+        "managedSourceDirectories" -> managedSourceDirectories.value.distinct
+          .mkString(File.pathSeparator),
+        "scalacOptions" -> scalacOptions.value.mkString(" "),
+        "classDirectory" -> classDirectory.value.getAbsolutePath,
+        "dependencyClasspath" -> dependencyClasspath.value
+          .map(_.data.toString)
+          .mkString(File.pathSeparator),
+        "scalaVersion" -> scalaVersion.value,
+        "sourceJars" -> {
+          val sourceJars = for {
+            configurationReport <- updateClassifiers.value.configurations
+            moduleReport <- configurationReport.modules
+            (artifact, file) <- moduleReport.artifacts
+            if artifact.classifier.exists(_ == "sources")
+          } yield file
+          sourceJars.mkString(File.pathSeparator)
+        }
+      )
+    }
+
+  def metalsWriteBuildInfoTask: Def.Initialize[Task[Unit]] = Def.task {
+    val props = new java.util.Properties()
+    metalsBuildInfoTask.value.foreach {
+      case (k, v) => props.setProperty(k, v)
+    }
+    val out = new ByteArrayOutputStream()
+    props.store(out, null)
+    val baseDir = baseDirectory.in(ThisBuild).value
+    val buildinfoDir = baseDir / ".metals" / "buildinfo" / thisProject.value.id
+    IO.delete(buildinfoDir)
+    buildinfoDir.mkdirs()
+    val outFile = buildinfoDir / s"${configuration.value.name}.properties"
+    IO.write(outFile, out.toString())
+    streams.value.log.info(s"Created ${outFile.relativeTo(baseDir).get}")
```

Comment:
Can we print out the absolute path to `outFile`? I just tested this locally and realized with relative paths you can't Cmd+click on the path in iTerm to open the file. I think we should make it easy to poke into the files for curious users.

Reply:
minor non-blocking comment on this too: Is `Created` always accurate? Don't these get `Updated` sometimes? If that's the case we could get away with `Written` :P

Reply:
> Wrote /path/to/foo

Reply:
😄👍 I agree. I noticed that sbt itself is very inconsistent in this regard. It says `Writing Ivy file ...xml`, but then `Wrote ...pom` 😬 . 

So "Written" or "Wrote" or "Writing"?

Reply:
Slightly longer suggestion

| Written build information for lsp4s/compile to /path/to/foo

Reply:
I think this is will be too long with absolute paths and quite repetitive:

```
Written build information for integration/compile to /Users/laughedelic/dev/laughedelic/metals/.metals/buildinfo/integration/compile.properties
```

Reply:
It's a bit long I agree. My intention was to vertically align the sbt projects/configurations so to easily spot what was included.

Reply:
The path prefix is the same, so they will be aligned. But you should also take into account that these logs are not ordered (sbt tasks aggregation runs in parallel).